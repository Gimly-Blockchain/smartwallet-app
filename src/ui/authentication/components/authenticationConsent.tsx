import React from 'react'
import { ButtonSection } from 'src/ui/structure/buttonSectionBottom'
import { Text, StyleSheet, View } from 'react-native'
import I18n from 'src/locales/i18n'
import strings from '../../../locales/strings'
import { Colors, Typography, Spacing } from 'src/styles'
import { IssuerCard } from '../../documents/components/issuerCard'
import { AuthenticationRequestSummary } from '../../../actions/sso/types'

interface Props {
  authenticationDetails: AuthenticationRequestSummary
  generateAndSendAuthenticationResponse: Function
  cancelAuthenticationRequest: Function
}

interface State {}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.backgroundLightMain,
  },
  topSection: {
    flex: 0.9,
  },
  issuerCard: {
    marginTop: Spacing.LG,
  },
  authRequestContainer: {
    flex: 1,
    paddingHorizontal: '10%',
    justifyContent: 'flex-start',
    alignItems: 'center',
    marginTop: Spacing.XL,
  },
  authRequestText: {
    ...Typography.baseFontStyles,
    ...Typography.centeredText,
    fontSize: Typography.textLG,
    marginTop: Spacing.SM,
  },
  buttonSection: {
    flex: 0.1,
  },
})

export class AuthenticationConsentComponent extends React.Component<
  Props,
  State
> {
  public state = {
    pending: false,
  }

  private handleConfirm = () => {
    this.setState({ pending: true })
    return this.props.generateAndSendAuthenticationResponse()
  }

  public render() {
    const { requester, description } = this.props.authenticationDetails
    return (
      <View style={styles.container}>
        <View style={styles.topSection}>
          <IssuerCard issuer={requester} style={styles.issuerCard} />
          <View style={styles.authRequestContainer}>
            <Text style={styles.authRequestText}>
              {I18n.t(strings.WOULD_YOU_LIKE_TO)}
            </Text>
            <Text
              style={[styles.authRequestText, { fontSize: Typography.text4XL }]}
            >
              {description}
            </Text>
            <Text style={styles.authRequestText}>
              {I18n.t(strings.WITH_YOUR_SMARTWALLET)}
            </Text>
          </View>
        </View>
        <View style={styles.buttonSection}>
          <ButtonSection
            disabled={this.state.pending}
            denyDisabled={this.state.pending}
            confirmText={I18n.t(strings.AUTHORIZE)}
            denyText={I18n.t(strings.DENY)}
            handleConfirm={this.handleConfirm}
            handleDeny={() => this.props.cancelAuthenticationRequest()}
          />
        </View>
      </View>
    )
  }
}
