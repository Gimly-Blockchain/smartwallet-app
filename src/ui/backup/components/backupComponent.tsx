import * as React from 'react'
import { ScrollView, StyleSheet, Text, View } from 'react-native'
import { Container } from '../../structure'
import { Colors } from '../../../styles'
import { fontMain } from '../../../styles/typography'
import { GradientButton } from '../../structure/gradientButton'
import { NavigationSection } from '../../structure/navigationSection'

interface Props {
  isLoading: boolean
  isAutoBackupEnabled: boolean
  enableAutoBackup: () => void
  onDisableAutoBackup: () => void
  exportBackup: () => void
  goBack: () => void
  lastBackup?: string
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.baseBlack,
    justifyContent: 'flex-start',
    alignItems: 'stretch',
  },
  scrollView: {
    padding: 18,
  },
  section: {
    backgroundColor: Colors.black,
    justifyContent: 'flex-start',
    alignItems: 'stretch',
    marginTop: 32,
    borderRadius: 8,
    padding: 24,
    paddingBottom: 28,
  },
  title: {
    fontFamily: fontMain,
    color: Colors.white,
    fontSize: 28,
  },
  subtitle: {
    fontFamily: fontMain,
    color: Colors.white,
    fontSize: 22,
    textAlign: 'center',
  },
  description: {
    color: Colors.white050,
    fontSize: 16,
    marginTop: 8,
    textAlign: 'center',
  },
  warning: {
    color: Colors.error,
    fontSize: 14,
    textAlign: 'center',
    marginTop: 28,
  },
  info: {
    textAlign: 'center',
    color: Colors.purpleMain,
    marginTop: 28,
  },
  buttonContainer: {
    marginTop: 22,
    width: '100%',
    paddingVertical: 9,
    margin: 0,
    borderRadius: 8,
  },
  buttonText: {
    fontSize: 18,
  },
})
const BackupComponent: React.FC<Props> = ({
  isLoading,
  enableAutoBackup,
  onDisableAutoBackup,
  exportBackup,
  isAutoBackupEnabled,
  goBack,
  lastBackup,
}) => {
  return (
    <Container style={styles.container}>
      <NavigationSection
        onNavigation={goBack}
        isBackButton={true}
        isDark={true}
      />
      <ScrollView style={styles.scrollView}>
        <Text style={styles.title}>Backup options</Text>

        <View style={styles.section}>
          <Text style={styles.subtitle}>Backup data on your own</Text>
          <Text style={styles.description}>
            Download an encrypted copy of the data in your SmartWallet to your
            device. Make sure to keep your backups up-to-date and stored
            somewhere safe
          </Text>
          <GradientButton
            containerStyle={styles.buttonContainer}
            textStyle={styles.buttonText}
            onPress={exportBackup}
            text={'Export file'}
            disabled={isLoading}
          />
          {lastBackup && (
            <Text style={styles.info}>
              {`Last backup ${new Date(lastBackup).toLocaleDateString()}`}
            </Text>
          )}
        </View>

        <View style={styles.section}>
          <Text style={styles.subtitle}>Jolocom backup service</Text>
          <Text style={styles.description}>
            Store your information using Jolocom’s backup service. With this
            option, the Jolocom team cannot view your data and your info stays
            automatically up-to-date.
          </Text>
          {isAutoBackupEnabled ? (
            <React.Fragment>
              <GradientButton
                containerStyle={styles.buttonContainer}
                textStyle={styles.buttonText}
                text={isLoading ? 'Deleting backup...' : 'Disable auto-backup'}
                onPress={onDisableAutoBackup}
                disabled={isLoading}
              />
              {lastBackup && (
                <Text style={styles.info}>
                  {`Last backup ${new Date(lastBackup).toLocaleDateString()}`}
                </Text>
              )}
            </React.Fragment>
          ) : (
            <React.Fragment>
              <GradientButton
                containerStyle={styles.buttonContainer}
                textStyle={styles.buttonText}
                onPress={enableAutoBackup}
                text={
                  isLoading ? 'Synchronization started…' : 'Use backup service'
                }
                disabled={isLoading}
              />
              <Text style={styles.warning}>
                By clicking here i agreed backing up my data on Jolocom’s
                central server and i know that it is not viewable by anyone
              </Text>
            </React.Fragment>
          )}
        </View>
      </ScrollView>
    </Container>
  )
}

export default BackupComponent
