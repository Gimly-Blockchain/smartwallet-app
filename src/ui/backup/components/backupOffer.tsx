import * as React from 'react'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import { Container } from '../../structure'
import { Colors } from '../../../styles'
import { GradientButton } from '../../structure/gradientButton'
import { ActionSheet } from '../../structure/actionSheet'

interface Props {
  enableAutoBackup: () => void
  manualBackup: () => void
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.baseBlack,
    justifyContent: 'flex-start',
    alignItems: 'stretch',
  },
  title: {
    color: Colors.white080,
    fontSize: 22,
    textAlign: 'center',
    marginHorizontal: 40,
  },
  logoTitle: {
    color: Colors.white080,
    fontSize: 22,
    textAlign: 'center',
    marginHorizontal: 108,
  },
  bottomSheet: {
    position: 'absolute',
    bottom: 0,
    backgroundColor: Colors.black,
    height: 230,
    width: '100%',
    borderRadius: 22,
    borderBottomLeftRadius: 0,
    borderBottomRightRadius: 0,
    paddingTop: 28,
    paddingHorizontal: 20,
  },
  button: {
    marginTop: 28,
    marginHorizontal: 20,
  },
  info: {
    color: Colors.white050,
    fontSize: 14,
    textAlign: 'center',
    marginTop: 10,
  },
  textButton: {
    color: Colors.white,
    fontSize: 20,
    textAlign: 'center',
    marginTop: 20,
  },
})
export const BackupOfferComponent: React.FC<Props> = ({
  enableAutoBackup,
  manualBackup,
}) => {
  return (
    <Container style={styles.container}>
      <Text style={styles.title}>
        Store your data by yourself or try to use Jolocom backup service!
      </Text>
      <View>{/* TODO add correct icon*/}</View>
      <Text style={styles.logoTitle}>Automatic synchronization</Text>
      <ActionSheet showSlide={true}>
        <GradientButton
          onPress={enableAutoBackup}
          text={'Turn on backup service'}
        />
        <Text style={styles.info}>
          By clicking here i agreed backing up my data on Jolocomâ€™s central
          server server and i know that it is not viewable by anyone
        </Text>
        <TouchableOpacity onPress={manualBackup}>
          <Text style={styles.textButton}>Keep it manually</Text>
        </TouchableOpacity>
      </ActionSheet>
    </Container>
  )
}
