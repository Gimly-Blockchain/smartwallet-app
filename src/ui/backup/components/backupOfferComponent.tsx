import * as React from 'react'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import { Container } from '../../structure'
import { Colors } from '../../../styles'
import { NavigationSection } from '../../structure/navigationSection'
import { GradientButton } from '../../structure/gradientButton'

interface Props {
  enableAutoBackup: () => void
  close: () => void
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.baseBlack,
    justifyContent: 'flex-start',
    alignItems: 'stretch',
  },
  title: {
    color: Colors.white080,
    fontSize: 22,
    textAlign: 'center',
    marginHorizontal: 40,
  },
  logoTitle: {
    color: Colors.white080,
    fontSize: 22,
    textAlign: 'center',
    marginHorizontal: 108,
  },
  bottomSheet: {
    position: 'absolute',
    bottom: 0,
    backgroundColor: Colors.black,
    height: 230,
    width: '100%',
    borderRadius: 22,
    borderBottomLeftRadius: 0,
    borderBottomRightRadius: 0,
    paddingTop: 28,
    paddingHorizontal: 20,
  },
  button: {
    marginTop: 28,
    marginHorizontal: 20,
  },
  info: {
    color: Colors.white050,
    fontSize: 14,
    textAlign: 'center',
    marginTop: 10,
  },
  textButton: {
    color: Colors.white,
    fontSize: 20,
    textAlign: 'center',
    marginTop: 20,
  },
})
const BackupOfferComponent: React.FC<Props> = ({ close, enableAutoBackup }) => {
  return (
    <Container style={styles.container}>
      <NavigationSection
        onNavigation={close}
        isBackButton={false}
        isDark={true}
      />
      <Text style={styles.title}>
        Store your data by yourself or try to use Jolocom backup service!
      </Text>
      <View>{/* TODO add correct icon*/}</View>
      <Text style={styles.logoTitle}>Automatic synchronization</Text>
      <View style={styles.bottomSheet}>
        <GradientButton
          onPress={enableAutoBackup}
          text={'Turn on backup service'}
        />
        <Text style={styles.info}>
          By clicking here i agreed backing up my data on Jolocomâ€™s central
          server and i know that it is not viewable by anyone
        </Text>
        {/*TODO add onPress*/}
        <TouchableOpacity>
          <Text style={styles.textButton}>Keep it manually</Text>
        </TouchableOpacity>
      </View>
    </Container>
  )
}

export default BackupOfferComponent
