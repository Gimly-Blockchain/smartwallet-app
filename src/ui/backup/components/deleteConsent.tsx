import { StyleSheet, Text, View } from 'react-native'
import { Colors } from '../../../styles'
import React from 'react'

const loaders = require('react-native-indicator')

interface Props {
  isLoading: boolean
  onCancel: () => void
  onAgree: () => void
}
const styles = StyleSheet.create({
  title: {
    color: Colors.white,
    fontSize: 26,
  },
  info: {
    color: Colors.white,
    fontSize: 14,
    marginTop: 8,
  },
  loaderContainer: {
    alignItems: 'center',
  },
  buttonSection: {
    marginTop: 20,
    flexDirection: 'row',
    justifyContent: 'flex-end',
  },
  button: {
    margin: 10,
    color: Colors.white,
    fontSize: 18,
  },
})
export const DeleteConsent: React.FC<Props> = ({
  isLoading,
  onCancel,
  onAgree,
}) => (
  <React.Fragment>
    <Text style={styles.title}>{'Disable auto-backup'}</Text>
    {isLoading ? (
      <View style={styles.loaderContainer}>
        <loaders.RippleLoader size={42} strokeWidth={2} color={Colors.beige} />
      </View>
    ) : (
      <React.Fragment>
        <Text style={styles.info}>
          {
            'By disabling this function you will erase all information from our server'
          }
        </Text>
        <View style={styles.buttonSection}>
          <Text style={styles.button} onPress={onCancel}>
            {'Cancel'}
          </Text>
          <Text style={styles.button} onPress={onAgree}>
            {'Agree'}
          </Text>
        </View>
      </React.Fragment>
    )}
  </React.Fragment>
)
