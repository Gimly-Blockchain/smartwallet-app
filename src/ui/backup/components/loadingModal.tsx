import React from 'react'
import { StyleSheet, View } from 'react-native'
import { Colors } from '../../../styles'
import Modal from 'react-native-modal'

interface Props {
  isVisible: boolean
  close?: () => void
}

const styles = StyleSheet.create({
  container: {
    borderRadius: 8,
    backgroundColor: '#444145',
    justifyContent: 'center',
    alignItems: 'stretch',
    padding: 24,
  },
  title: {
    color: Colors.white,
    fontSize: 26,
    textAlign: 'center',
  },
  loaderContainer: {
    margin: 24,
  },
  waring: {
    color: Colors.error,
    fontSize: 14,
    textAlign: 'center',
  },
})
export const LoadingModal: React.FC<Props> = ({
  isVisible,
  close,
  children,
}) => (
  <Modal
    isVisible={isVisible}
    animationIn={'fadeIn'}
    animationOut={'fadeOut'}
    onBackdropPress={close}
  >
    <View style={styles.container}>{children}</View>
  </Modal>
)
