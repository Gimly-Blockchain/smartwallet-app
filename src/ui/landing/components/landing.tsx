import React, { ReactNode } from 'react'
import { StyleSheet, Dimensions } from 'react-native'
import { Button } from 'react-native-material-ui'
import { Block, Container, CenteredText } from 'src/ui/structure'
import { JolocomTheme } from 'src/styles/jolocom-theme'
import I18n from 'src/locales/i18n'
import { Landing00, Landing01, Landing02, Landing03 } from 'src/resources'
import strings from 'src/locales/strings'

const Carousel = require('react-native-snap-carousel').default
const Pagination = require('react-native-snap-carousel').Pagination

interface State {
  activeSlide: number
}

interface Props {
  handleGetStarted: () => void
  handleRecover: () => void
}

interface Slide {
  svgImage: ReactNode
  title: string
  infoText: string
}

const viewWidth: number = Dimensions.get('window').width
const headerFontSize =
  viewWidth < 360
    ? JolocomTheme.landingHeaderFontSizeSmall
    : JolocomTheme.landingHeaderFontSize
const labelFontSize =
  viewWidth < 360 ? JolocomTheme.labelFontSizeSmall : JolocomTheme.labelFontSize

const styles = StyleSheet.create({
  mainContainerStyle: {
    paddingTop: 0,
    backgroundColor: '#05050d',
    justifyContent: 'flex-end',
    flexDirection: 'column',
    flex: 1,
  },
  carouselTextContainer: {
    paddingHorizontal: viewWidth / 18,
    flex: 0.4,
    marginTop: 'auto',
    justifyContent: 'flex-end',
    backgroundColor: 'transparent',
  },
  activeDotStyle: {
    width: 8,
    height: 8,
    backgroundColor: JolocomTheme.dotColorActive,
  },
  inactiveDotStyle: {
    width: 4,
    height: 4,
    opacity: 0.6,
    backgroundColor: JolocomTheme.dotColorInactive,
  },
  header: {
    color: JolocomTheme.primaryColorSand,
    fontFamily: JolocomTheme.contentFontFamily,
    fontSize: headerFontSize,
    fontWeight: '100',
  },
  subHeader: {
    color: JolocomTheme.primaryColorSand,
    opacity: 0.8,
    fontFamily: JolocomTheme.contentFontFamily,
    fontSize: labelFontSize,
    fontWeight: '100',
    lineHeight: labelFontSize + 4,
    marginTop: 15,
  },
  paginationBlock: {
    flex: 0.15,
    backgroundColor: '#05050d',
  },
  buttonBlock: {
    flex: 0.3,
    backgroundColor: '#05050d',
  },
  buttonContainer: {
    marginTop: 5,
    height: 48,
    minWidth: 164,
    borderRadius: 4,
    backgroundColor: JolocomTheme.primaryColorPurple,
  },
  buttonText: {
    paddingVertical: 15,
    fontFamily: JolocomTheme.contentFontFamily,
    color: JolocomTheme.primaryColorWhite,
    fontSize: JolocomTheme.headerFontSize,
    fontWeight: '100',
    textAlign: 'center',
    minWidth: 158,
  },
})

const carouselInfo: Slide[] = [
  {
    svgImage: <Landing00 />,
    title: I18n.t(strings.YOUR_JOLOCOM_WALLET),
    infoText:
      I18n.t(
        strings.TAKE_BACK_CONTROL_OF_YOUR_DIGITAL_SELF_AND_PROTECT_YOUR_PRIVATE_DATA_AGAINST_UNFAIR_USAGE,
      ) + '.',
  },
  {
    svgImage: <Landing01 height={'100%'} width={'100%'} />,
    title: I18n.t(strings.ITS_EASY),
    infoText:
      I18n.t(strings.FORGET_ABOUT_LONG_FORMS_AND_REGISTRATIONS) +
      '. ' +
      I18n.t(
        strings.INSTANTLY_ACCESS_SERVICES_WITHOUT_USING_YOUR_SOCIAL_MEDIA_PROFILES,
      ) +
      '.',
  },
  {
    svgImage: <Landing03 height={'100%'} width={'100%'} />,
    title: I18n.t(strings.ENHANCED_PRIVACY),
    infoText:
      I18n.t(strings.SHARE_ONLY_THE_INFORMATION_A_SERVICE_REALLY_NEEDS) +
      '. ' +
      I18n.t(strings.PROTECT_YOUR_DIGITAL_SELF_AGAINST_FRAUD) +
      '.',
  },
  {
    svgImage: <Landing02 height={'100%'} width={'100%'} />,
    title: I18n.t(strings.GREATER_CONTROL),
    infoText:
      I18n.t(
        strings.KEEP_ALL_YOUR_DATA_WITH_YOU_IN_ONE_PLACE_AVAILABLE_AT_ANY_TIME,
      ) +
      '. ' +
      I18n.t(strings.TRACK_WHERE_YOU_SIGN_IN_TO_SERVICES) +
      '.',
  },
]

export class LandingComponent extends React.Component<Props, State> {
  state = {
    activeSlide: 0,
  }

  private renderItem = ({ item }: { item: Slide }) => {
    const { svgImage, title, infoText } = item
    return (
      <Block>
        {svgImage}
        <Block style={styles.carouselTextContainer}>
          <CenteredText style={styles.header} msg={title} />
          <CenteredText style={styles.subHeader} msg={infoText} />
        </Block>
      </Block>
    )
  }

  private renderPagination() {
    const { activeSlide } = this.state
    return (
      <Pagination
        dotsLength={carouselInfo.length}
        activeDotIndex={activeSlide}
        dotStyle={styles.activeDotStyle}
        inactiveDotStyle={styles.inactiveDotStyle}
        inactiveDotOpacity={0.4}
        inactiveDotScale={0.6}
      />
    )
  }

  render() {
    return (
      <Container style={styles.mainContainerStyle}>
        <Block>
          <Carousel
            data={carouselInfo}
            renderItem={this.renderItem}
            lockScrollWhileSnapping
            lockScrollTimeoutDuration={1000}
            loop
            sliderWidth={viewWidth}
            itemWidth={viewWidth}
            layout={'default'}
            onSnapToItem={(index: number) =>
              this.setState({ activeSlide: index })
            }
          />
        </Block>
        <Block style={styles.paginationBlock}>{this.renderPagination()}</Block>
        <Block style={styles.buttonBlock}>
          <Button
            raised
            onPress={this.props.handleGetStarted}
            style={{
              container: styles.buttonContainer,
              text: styles.buttonText,
            }}
            upperCase={false}
            text={I18n.t(strings.GET_STARTED)}
          />
          <Button
            raised
            onPress={this.props.handleRecover}
            style={{
              container: styles.buttonContainer,
              text: styles.buttonText,
            }}
            upperCase={false}
            text={'Recover Identity'}
          />
        </Block>
      </Container>
    )
  }
}
