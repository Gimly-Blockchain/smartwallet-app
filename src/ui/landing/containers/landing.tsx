import React from 'react'
import { connect } from 'react-redux'
import { LandingComponent } from 'src/ui/landing/components/landing'
import { registrationActions } from 'src/actions/'
import { withErrorHandling } from '../../../actions/modifiers'
import { showErrorScreen } from '../../../actions/generic'
import { AppError, ErrorCode } from '../../../lib/errors'
import { routeList } from '../../../routeList'
import { ThunkDispatch } from '../../../store'
import { StatusBar } from 'react-native'

interface Props extends ReturnType<typeof mapDispatchToProps> {}

export class LandingContainer extends React.Component<Props> {
  render() {
    return (
      <React.Fragment>
        <StatusBar barStyle="light-content" />
        <LandingComponent handleButtonTap={this.props.startRegistration} />
      </React.Fragment>
    )
  }
}

const mapDispatchToProps = (dispatch: ThunkDispatch) => ({
  startRegistration: () =>
    dispatch(
      withErrorHandling(
        showErrorScreen,
        (err: AppError) =>
          new AppError(ErrorCode.RegistrationFailed, err, routeList.Landing),
      )(registrationActions.startRegistration),
    ),
})

export const Landing = connect(
  null,
  mapDispatchToProps,
)(LandingContainer)
