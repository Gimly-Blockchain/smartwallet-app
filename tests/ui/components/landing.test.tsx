import React from 'react'
import { LandingComponent } from 'src/ui/landing/components/landing'
import { shallow } from 'enzyme'

describe('landing component', () => {
  it('matches the snapshot on render', () => {
    const props = {
      handleGetStarted: jest.fn(),
      handleRecover: jest.fn(),
    }

    const rendered = shallow(<LandingComponent {...props} />)
    expect(rendered).toMatchSnapshot()
  })
})
