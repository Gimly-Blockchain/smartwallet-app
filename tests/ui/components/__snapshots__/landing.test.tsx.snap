// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landing component matches the snapshot on render 1`] = `
<Container
  style={
    Object {
      "backgroundColor": "rgb(5, 5, 13)",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-end",
      "paddingTop": 0,
    }
  }
>
  <Block>
    <Carousel
      activeAnimationOptions={null}
      activeAnimationType="timing"
      activeSlideAlignment="center"
      activeSlideOffset={20}
      apparitionDelay={0}
      autoplay={false}
      autoplayDelay={1000}
      autoplayInterval={3000}
      callbackOffsetMargin={5}
      containerCustomStyle={Object {}}
      contentContainerCustomStyle={Object {}}
      data={
        Array [
          Object {
            "infoText": "Take back control of your digital self and protect your private data against unfair usage.",
            "svgImage": <Landing00 />,
            "title": "Your Jolocom Wallet",
          },
          Object {
            "infoText": "Forget about long forms and registrations. Instantly access services without using your social media profiles.",
            "svgImage": <Landing00
              height="100%"
              width="100%"
            />,
            "title": "It's easy",
          },
          Object {
            "infoText": "Share only the information a service really needs. Protect your digital self against fraud.",
            "svgImage": <Landing00
              height="100%"
              width="100%"
            />,
            "title": "Enhanced privacy",
          },
          Object {
            "infoText": "Keep all your data with you in one place, available at any time. Track where you sign in to services.",
            "svgImage": <Landing00
              height="100%"
              width="100%"
            />,
            "title": "Greater control",
          },
        ]
      }
      enableMomentum={false}
      enableSnap={true}
      firstItem={0}
      hasParallaxImages={false}
      inactiveSlideOpacity={0.7}
      inactiveSlideScale={0.9}
      inactiveSlideShift={0}
      itemWidth={750}
      layout="default"
      lockScrollTimeoutDuration={1000}
      lockScrollWhileSnapping={true}
      loop={true}
      loopClonesPerSide={3}
      onSnapToItem={[Function]}
      renderItem={[Function]}
      scrollEnabled={true}
      shouldOptimizeUpdates={true}
      slideStyle={Object {}}
      sliderWidth={750}
      swipeThreshold={20}
      useScrollView={false}
      vertical={false}
    />
  </Block>
  <Block
    style={
      Object {
        "flex": 0.15,
      }
    }
  >
    <Pagination
      activeDotIndex={0}
      dotStyle={
        Object {
          "backgroundColor": "rgb(255, 222, 188)",
          "height": 8,
          "width": 8,
        }
      }
      dotsLength={4}
      inactiveDotOpacity={0.4}
      inactiveDotScale={0.6}
      inactiveDotStyle={
        Object {
          "backgroundColor": "rgba(255, 254, 252, 0.7)",
          "height": 4,
          "width": 4,
        }
      }
      tappableDots={false}
      vertical={false}
    />
  </Block>
  <Block
    style={
      Object {
        "flex": 0.1,
      }
    }
  >
    <JolocomButton
      onPress={[MockFunction]}
      raised={true}
      text="Get started"
      upperCase={false}
    />
  </Block>
</Container>
`;
